app/code/Magento/Cms/Model/ResourceModel/Block/Grid/Collection.php

```diff
@@ -91,13 +91,34 @@ public function setAggregations($aggregations)
      * @param int $limit
      * @param int $offset
      * @return array
+     * @throws \Magento\Framework\Exception\LocalizedException
      */
     public function getAllIds($limit = null, $offset = null)
     {
         return $this->getConnection()->fetchCol($this->_getAllIdsSelect($limit, $offset), $this->_bindParams);
     }
 
     /**
+     * Create all ids retrieving select with limitation
+     *
+     * @param int $limit
+     * @param int $offset
+     * @return \Magento\Framework\DB\Select
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    protected function _getAllIdsSelect($limit = null, $offset = null)
```

Comment:
Actually looking at how pages work the problem could be that this class still has the function `getAllIds` rather than just using the parent method. Would also be nice to remove these methods if we can rely on the parent methods.